File: Python3/prime.py
# Looking for largest prime number
def faktor_prima(n):
    factor = 2
    last_factor = 1

    while factor * factor <= n:
        if n % factor == 0:
            last_factor = factor
            n //= factor
            while n % factor == 0:
                n //= factor
        factor += 1 if factor == 2 else 2

    return n if n > 1 else last_factor

num = int(input("Masukkan sembarang angka: "))
# as example is 600851475143
print(faktor_prima(num))

File: Julia/prime.jl
# Looking for largest prime number
function faktor_prima(n)
    f = 2           #factor
    l_f = 1         #last factor

    while f * f <= n
        if n % f == 0
            l_f = f
            n /= f
            while n % f == 0
                n /= f
            end
        end
        f = f == 2 ? 3 : f + 2
    end
    return n > 1 ? n : l_f
end

println("Masukkan sembarang angka: ")
num = parse(Int, readline())
# as example is 600851475143
println("Faktor primat terbesarnya adalah: ", faktor_prima(num))
